llvm/lib/Target/LoongArch/LoongArchISelLowering.cpp

```diff
@@ -937,6 +937,9 @@ LoongArchTargetLowering::lowerGlobalTLSAddress(SDValue Op,
   GlobalAddressSDNode *N = cast<GlobalAddressSDNode>(Op);
   assert(N->getOffset() == 0 && "unexpected offset in global node");
 
+  if (DAG.getTarget().useEmulatedTLS())
+    DAG.getContext()->emitError("the emulated TLS is prohibited.");
```

Comment:
https://llvm.org/docs/CodingStandards.html#error-and-warning-messages no trailing period :)

Reply:
thanks!